pkg/controller/infinispan/util/cluster.go

```diff
@@ -198,6 +258,88 @@ func (c Cluster) CreateCache(cacheName, cacheXml, secretName, podName, namespace
 	return nil
 }
 
+// CreateCacheWithAuth create cluster cache on the pod `podName`
+func (c Cluster) CreateCacheWithAuth(user, pass, cacheName, cacheXML, podName, namespace, protocol string) error {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return err
+	}
+
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches/%s", protocol, podIP, cacheName)
+	commands := []string{"curl",
+		"--insecure",
+		"--http1.1",
+		"-w", "\n%{http_code}", // add http status at the end
+		"-d", fmt.Sprintf("%s", cacheXML),
+		"-H", "Content-Type: application/xml",
+		"-u", fmt.Sprintf("operator:%v", pass),
+		"-X", "POST",
+		httpURL,
+	}
+
+	execOptions := ExecOptions{Command: commands, PodName: podName, Namespace: namespace}
+	execOut, execErr, err := c.Kubernetes.ExecWithOptions(execOptions)
+	if err != nil {
+		return fmt.Errorf("unexpected error creating cache, stderr: %v, err: %v", execErr, err)
+	}
+
+	// Split lines in standard output, HTTP status code will be last
+	execOutLines := strings.Split(execOut.String(), "\n")
+
+	httpCode, err := strconv.ParseUint(execOutLines[len(execOutLines)-1], 10, 64)
+	if err != nil {
+		return err
+	}
+
+	if httpCode > 299 || httpCode < 200 {
+		return fmt.Errorf("server side error creating cache: %s", execOut.String())
+	}
+
+	logger := log.WithValues("Request.Namespace", namespace, "Pod.Name", podName)
+	logger.Info("create cache completed successfully", "http code", httpCode)
+	return nil
+}
+
+// CreateCacheWithTemplateName create cluster cache on the pod `podName`
+func (c Cluster) CreateCacheWithTemplateName(user, pass, cacheName, templateName, podName, namespace, protocol string) error {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return err
+	}
+
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches/%s?template=%s", protocol, podIP, cacheName, templateName)
```

Comment:
@rigazilla, looks like we need to check which error will display if template is not exists or incorrect otherwise check with 
GET /rest/v2/cache-managers/{cacheManagerName}/cache-configs/templates before
or we can keep and show available templates list before

Reply:
the ISPN error is reported in the err object and logged by the operator.
I would leave improved error handling in a next PR, maybe we want to show error also in the cr status.